frontend/src/components/professor/GameManagement/ActivityDetails/ActivityRequirements/formHelpers.js

```diff
@@ -0,0 +1,44 @@
+import { RequirementType } from '../../../../../utils/constants'
+
+export const onInputChange = (requirementId, value, setRequirementsList) => {
+  setRequirementsList((prevState) =>
+    prevState.map((requirement) => (requirement.id === requirementId ? { ...requirement, answer: value } : requirement))
+  )
+}
+
+export const onCheckboxChange = (requirementId, setRequirementsList) => {
+  setRequirementsList((prevState) =>
+    prevState.map((requirement) =>
+      requirement.id === requirementId ? { ...requirement, selected: !requirement.selected } : requirement
+    )
+  )
+}
+
+export const onSwitchChange = (requirementId, setRequirementsList) => {
+  setRequirementsList((prevState) =>
+    prevState.map((requirement) =>
+      requirement.id === requirementId
+        ? { ...requirement, value: !requirement.value, answer: !requirement.value }
+        : requirement
+    )
+  )
+}
+
+export const onSelectChange = (requirementId, value, setRequirementsList) => {
+  setRequirementsList((prevState) =>
+    prevState.map((requirement) => (requirement.id === requirementId ? { ...requirement, answer: value } : requirement))
+  )
+}
+
```

Comment:
NIt, i w sumie nieistotne więc możesz pominąć - nie da się wydzielić większości kodu z tych czterech funkcji wyżej do jednej wspólnej?

Reply:
Na razie pomińmy to, jeżeli to nie problem. W przyszłości zrobimy refactor